packages/netsuite-adapter/src/deletion_calculator.ts

```diff
@@ -0,0 +1,151 @@
+    return []
+  }
+
+  const customRecordElemNames = new Set(serviceTargetedCustomRecords
+    .map(instanceElement => instanceElement.elemID.getFullName()))
+
+  return currentElements
+    .filter(item => item.elemID.idType === 'instance')
+    .filter(item => !customRecordElemNames.has(item.elemID.getFullName()))
+    .map(item => item.elemID)
+}
+
+const calculateDeletedCustomRecordsFromService = async (
+  client: NetsuiteClient,
+  currentElements: ElemServiceID[],
+  query: NetsuiteQuery,
+  serviceTargetedCustomRecords: InstanceElement[],
+): Promise<ElemID[]> => {
+  const customRecordTypesToIgnore = new Set(serviceTargetedCustomRecords.map(item => item.elemID.typeName))
+  const serviceCustomRecordCounters = await getCustomRecordCounters(
+    client,
+    { isCustomRecordTypeMatch: query.isCustomRecordTypeMatch },
+    customRecordTypesToIgnore,
+  )
+  const currentCustomRecords = _(currentElements)
+    .filter(item => !customRecordTypesToIgnore.has(item.elemID.typeName))
+    .groupBy(item => item.elemID.typeName)
+    .value()
+
+  const customRecordTypesWithDeletions = Object.values(currentCustomRecords)
+    .filter(instances => instances.length !== serviceCustomRecordCounters.get(instances[0].serviceID))
+    .flat()
+
+  const serviceCustomRecords = await Promise.all(
+    customRecordTypesWithDeletions.map(async customRecordType => {
+      const instancesScriptIds = await getCustomRecordTypeInstances(client, customRecordType.elemID.typeName)
+      return {
+        customRecordType: customRecordType.elemID.typeName,
+        instancesScriptIds: new Set(instancesScriptIds),
+      }
+    })
+  )
+  const serviceCustomRecordInstancesMap = new Map(
+    serviceCustomRecords.map(obj => [obj.customRecordType, obj.instancesScriptIds])
+  )
+
+  return Object.values(currentCustomRecords).flat()
+    .filter(item => !serviceCustomRecordInstancesMap.get(item.elemID.typeName)?.has(item.serviceID))
+    .map(item => item.elemID)
+}
+
+const calculateDeletedCustomRecords = async (
+  client: NetsuiteClient,
+  currentElements: ElemServiceID[],
+  query: NetsuiteQuery,
+  serviceTargetedCustomRecords: InstanceElement[],
+): Promise<ElemID[]> => {
+  if (currentElements.length === 0) {
+    return []
+  }
+
+  const deletedCustomRecords = calculateDeletedCustomRecordsFromTarget(
+    currentElements,
+    serviceTargetedCustomRecords,
+  )
+
+  const deletedCustomRecordsAdditional = await calculateDeletedCustomRecordsFromService(
+    client,
+    currentElements,
+    query,
+    serviceTargetedCustomRecords,
+  )
+  return deletedCustomRecords.concat(deletedCustomRecordsAdditional)
+}
+
+export const getDeletedElements = async (
+  client: NetsuiteClient,
+  elementsSourceIndex: LazyElementsSourceIndexes,
+  query: NetsuiteQuery,
+  serviceInstancesIds: ObjectID[],
+  serviceTargetedCustomRecords: InstanceElement[],
+): Promise<ElemID[]> => {
+  const currentElements = await getCurrentScriptIds(elementsSourceIndex)
```

Comment:
maybe I should split this list into `isTypeMatch` and `isCustomRecordTypeMatch` in one run (using groupBy?)

Reply:
`isTypeMatch` & `isCustomRecordTypeMatch` are working on the adapter config inclusions&exclusions, without checking if the type is a custom record type or a standard type. So I think that we should first split into types/custom records types-
```suggestion
  const [currentInstances, currentCustomRecords] = _.partition(
    await getCurrentScriptIds(elementsSourceIndex),
    item => isStandardTypeName(item.elemID.typeName) || item.elemID.idType === 'type' // for custom record types
  )
```